src/web/app/components/question-submission-form/question-submission-form.component.ts

```diff
@@ -197,4 +196,19 @@ export class QuestionSubmissionFormComponent implements OnInit {
     return this.feedbackResponseService.isFeedbackResponseDetailsEmpty(
         this.model.questionType, responseDetails);
   }
+
+  /**
+   * Updates validity of all responses in a question.
+   */
+  updateValidity(isValid: boolean): void {
+    const recipientSubmissionForms: FeedbackResponseRecipientSubmissionFormModel[] =
+        this.model.recipientSubmissionForms.slice();
```

Comment:
```suggestion
        this.model.recipientSubmissionForms.slice().map((model) => Object.assign({}, model, {isValid}))
```

This might be what you want.